pylds/vector_fields.py

```diff
@@ -181,5 +181,99 @@ def forcing(t, u, perturbation_params = [1, 0.15, 0.5]):
     
     return perturbation
 
+def HenonHeiles_vector_field(t, u, PARAMETERS = None):
+    """
+    Returns 2D Henon-Heiles vector field at time t, for an array of points in phase space.
+    Number of model parameters: 0 . PARAMETERS = [wx, wy, delta]
+    Functional form: v = (x wx^2 + 2xy, y wy^2 + x^2 - delta y^2), with u = (x, y)
+    
+    Parameters
+    ----------
+    t : float
+        fixed time-point of vector field, for all points in phase space.
+        
+    u : array_like, shape(n,)
+        points in phase space to determine vector field at time t.
+        
+    PARAMETERS : list of floats
+        vector field parameters
+    
+    Returns
+    -------
+    v : array_like, shape(n,)
+        vector field corresponding to points u, in phase space at time t
+    """
+    N_dof = u.shape[-1]
+    points_positions = u.T[:int(N_dof/2)]
+    points_momenta = u.T[int(N_dof/2):]
+    x, y = points_positions
+    p_x, p_y = points_momenta 
+    
+    # Hamiltonian Model Parameter
+    # None
+    
+    # Vector field defintion
+    v_x   =  p_x
+    v_y   =  p_y
+    v_p_x = -x - 2*x*y
+    v_p_y = -x**2 -y + y**2
+    v = np.array([v_x, v_y, v_p_x, v_p_y]).T
+    return v
+
+def HenonHeiles_potential(positions, PARAMETERS = None):
+    x, y = positions.T
+    # Function parameters
+    # None
+    # Potential energy function
+    V = (1/2)*(x**2 + y**2) + (y * x**2) - (1/3)*y**3
+    return V
+
+def SaddleNF_vector_field(t, u, PARAMETERS = None):
+    """
+    Returns 2D Henon-Heilles vector field at time t, for an array of points in phase space.
+    Number of model parameters: 0 . PARAMETERS = [wx, wy, delta]
+    Functional form: v = (x wx^2 + 2xy, y wy^2 + x^2 - delta y^2), with u = (x, y)
```

Comment:
Correct me if I'm mistaken, but the system is the normal form of a saddle x centre, not a saddle. Also the docstring needs adjusting.